src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -384,6 +402,43 @@ private static string GetNextMethodName(ArrayBuilder<MethodSymbol> builder)
                             !IsDisplayClassParameter(parameter))
                         {
                             itemsAdded.Add(parameterName);
+
+                            // Display class variables created for method parameters shadow the parameters.
+                            if (_displayClassVariables.TryGetValue(parameterName, out var variable) && variable.Kind == DisplayClassVariableKind.Parameter)
+                            {
+                                int saveCount = methodBuilder.Count;
+                                int localIndex = 0;
+                                foreach (var local in _localsForBindingOutside)
```

Comment:
>_localsForBindingOutside [](http://example.com/codeflow?start=54&length=24)

Consider using `_localsForBindingOutside.Concat(_localsForBindingInside)` to simplify.